pulsar-functions/proto/src/main/proto/Function.proto

```diff
@@ -70,6 +70,7 @@ message SourceSpec {
     SubscriptionType subscriptionType = 3;
     map<string,string> topicsToSerDeClassName = 4;
     uint64 timeoutMs = 6;
+    string topicsPattern = 7;
```

Comment:
Shouln't it be a map<string, string> topicsPatternToSerDeClassName?

Reply:
I thought about it earlier but then I feel we can combine it in `topicsToSerDeClassName` only. in one of the usecase, topicPattern can be used to start all consumers under a given namespace. now, there is a possibility that some of the topics may need different deserializers so, PulsarSource can first check deserializer for a specific topic in the map and then try to match the pattern in the same map. in this way, we can save cpu by avoiding pattern-matching on every message and will be easier for user to configure it in function-config `topicToSerDeMap.put("pattern", deserializer)`

```
+            SerDe deserializer = null;
+            if (this.topicToSerDeMap.containsKey(topicName)) {
+                deserializer = this.topicToSerDeMap.get(topicName);
+            } else if (isTopicsPattern) {
+                deserializer = this.topicToSerDeMap.get(this.pulsarSourceConfig.getTopicsPattern());
+            }
```